be/src/exec/csv_scanner.cpp

```diff
@@ -506,4 +514,45 @@ void CSVScanner::_report_rejected_record(const CSVReader::Record& record, const
     _state->append_rejected_record_to_file(record.to_string(), err_msg, _curr_reader->filename());
 }
 
+static TypeDescriptor get_type_desc(const Slice& field) {
+    StringParser::ParseResult result;
+
+    StringParser::string_to_int<int64_t>(field.get_data(), field.get_size(), &result);
+    if (result == StringParser::PARSE_SUCCESS) {
+        return TypeDescriptor(TYPE_BIGINT);
+    }
+
+    StringParser::string_to_float<double>(field.get_data(), field.get_size(), &result);
+    if (result == StringParser::PARSE_SUCCESS) {
+        return TypeDescriptor(TYPE_DOUBLE);
+    }
+
+    StringParser::string_to_bool(field.get_data(), field.get_size(), &result);
+    if (result == StringParser::PARSE_SUCCESS) {
+        return TypeDescriptor(TYPE_BOOLEAN);
+    }
+
+    // default VARCHAR.
+    return TypeDescriptor::create_varchar_type(TypeDescriptor::MAX_VARCHAR_LENGTH);
+}
+
+Status CSVScanner::get_schema(std::vector<SlotDescriptor>* schema) {
+    if (schema == nullptr) {
+        return Status::InternalError("ouput schema is null");
+    }
+
+    RETURN_IF_ERROR(_init_reader());
+
+    CSVReader::Record record;
+    RETURN_IF_ERROR(_curr_reader->next_record(&record));
+
+    CSVReader::Fields fields;
+    _curr_reader->split_record(record, &fields);
+    for (size_t i = 0; i < fields.size(); i++) {
+        // column name: $1, $2, $3...
+        schema->emplace_back(SlotDescriptor(i, fmt::format("${}", i + 1), get_type_desc(fields[i])));
```

Comment:
Put the magic const string into the csv header file, like  DEFAULT_CSV_COLUMN_NAME_FORMAT = "${}" ?

Reply:
And, where is the code to merge several sample file schema?

Reply:
https://github.com/StarRocks/starrocks/pull/36892